### PR TITLE
Database

### PR DESCRIPTION
**Pull request by matthewdking**
relates #82 

- [x] Creates a database on heroku
- [x] Inputs seed data to database when running db_build
- [x] Connect to database when app opens
- [x] Redux store with data in
- [x] Persist store for all pages

### CODE REVIEW

File: database/seed_data.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+const Career = require('./career_model.js');
+
+const careers = [
+  new Career({ title: 'Doctor',
+    tagline: 'Promots and Improves Health',
+    description: 'A doctor is a professional who practises medicine, which is concerned with promoting, maintaining, or restoring health.',
+    image: 'https://cdn4.iconfinder.com/data/icons/avatar-male/920/Accountant_Character-512.png',
+    degree: 'Medicine',
+    grade_bagrut: '11.10',
```
Comment (veredrec) [inline database/seed_data.js]: The grades are `Number` in the Schema but `String` in the database content :confused:

Comment (matthewdking) [reply database/seed_data.js]: Its in quotes here so its a string. However I can change them both to number?

Comment (matthewdking) [reply database/seed_data.js]: Yeah you're right I will change it!

File: src/store.config.js [hunk 2]
```diff
@@ -1,72 +1,14 @@
-import { createStore } from 'redux';
+import thunk from 'redux-thunk';
```
Comment (Heathercoraje) [inline src/store.config.js]: what does this do?

File: .babelrc [hunk 3]
```diff
@@ -1,3 +1,4 @@
 {
-  "presets": ["env", "react"]
+  "presets": ["env", "react"],
+  "plugins": ["transform-object-rest-spread"]
```
Comment (Karyum) [inline .babelrc]: not going to request to remove this cause you still have an error, it's a pain to download and setup again.
which hopefully you guys would fix tomorrow and after it ( bug fixing days )

File: database/db_build.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+// This file should be run once to populate the database with seed data!!
+
+const Career = require('./career_model.js');
```
Comment (Karyum) [inline database/db_build.js]: what is this file for ??

Comment (Karyum) [reply database/db_build.js]: I see this file only being used in development not production.
so it's more of a development tool to populate the database.
best solution is to .gitignore it or if you have something else please comment

File: database/db_connection.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const mongoose = require('mongoose');
+require('env2')('./config.env');
+
+const db = mongoose.connection;
```
Comment (Karyum) [inline database/db_connection.js]: :arrow_up:

File: database/seed_data.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
+const Career = require('./career_model.js');
```
Comment (Karyum) [inline database/seed_data.js]: :arrow_up:

File: package.json [hunk 7]
```diff
@@ -35,16 +35,25 @@
...
     "lodash": "^4.17.4",
+    "mongoose": "^4.12.2",
     "prop-types": "^15.6.0",
     "react": "^16.0.0",
     "react-dom": "^16.0.0",
     "react-icons": "^2.2.7",
     "react-redux": "^5.0.6",
     "react-router-dom": "^4.2.2",
     "redux": "^3.7.2",
+    "redux-async-initial-state": "^0.2.1",
```
Comment (Karyum) [inline package.json]: is this being used anywhere?

File: src/containers/career.jsx [hunk 8]
```diff
@@ -24,14 +24,49 @@ class Career extends Component {
 
     this.state = {
       currentIndex: 0,
-      careers: []
+      careers: [
```
Comment (Karyum) [inline src/containers/career.jsx]: in the future it's better to have all of your states in redux not in the component state.
but it's fine for now

File: src/containers/career.jsx [hunk 9]
```diff
@@ -24,14 +24,49 @@ class Career extends Component {
...
     this.flipCard = this.flipCard.bind(this);
     this.flipCardBack = this.flipCardBack.bind(this);
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.careers !== this.props.careers) {
+      if (localStorage.liked) {
+        const likedCareers = JSON.parse(localStorage.liked);
+        const allCareers = nextProps.careers;
+        const notLikedCareers = _.differenceBy(allCareers, likedCareers, 'title');
```
Comment (Karyum) [inline src/containers/career.jsx]: i'm not big on lodash can you explain what this does?

File: src/reducers/data.reducer.js [hunk 10]
```diff
@@ -1,3 +1,10 @@
-const DataReducer = (state = []) => state;
+const DataReducer = (state = [], action) => {
+  switch (action.type) {
```
Comment (Karyum) [inline src/reducers/data.reducer.js]: you could delete this reducer

File: src/reducers/index.js [hunk 11]
```diff
@@ -1,8 +1,9 @@
 import { combineReducers } from 'redux';
 import DataReducer from './data.reducer';
```
Comment (Karyum) [inline src/reducers/index.js]: also delete this line

File: src/server.js [hunk 12]
```diff
@@ -1,13 +1,28 @@
...
 
 app.use(express.static(path.resolve(__dirname, '..', 'dist')));
 app.use(express.static('public'));
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+
+app.get('/api/careers', (req, res) => {
+  Careers.find({}, (error, result) => {
+    if (error) {
+      console.log(error);
```
Comment (Karyum) [inline src/server.js]: raise it as an issue that you should do better error handling

File: src/store.config.js [hunk 13]
```diff
@@ -1,72 +1,18 @@
-import { createStore } from 'redux';
+import * as asyncInitialState from 'redux-async-initial-state';
+import axios from 'axios';
```
Comment (Karyum) [inline src/store.config.js]: both of those lines here not usre you need them anymore

File: webpack.config.js [hunk 14]
```diff
@@ -40,5 +40,12 @@ module.exports = {
   ],
   resolve: {
     extensions: ['.js', '.jsx']
+  },
+  node: {
+    console: false,
+    net: 'empty',
+    dns: 'empty',
+    tls: 'empty',
+    fs: 'empty'
```
Comment (Karyum) [inline webpack.config.js]: what is this again?

File: database/db_build.js (hunk 4 above)
Comment (matthewdking) [reply database/db_build.js]: I agree :+1:

File: database/db_connection.js (hunk 5 above)
Comment (matthewdking) [reply database/db_connection.js]: this is used in start_server.js

File: database/seed_data.js (hunk 6 above)
Comment (matthewdking) [reply database/seed_data.js]: ignored  :hear_no_evil:

File: src/containers/career.jsx (hunk 9 above)
Comment (matthewdking) [reply src/containers/career.jsx]: google it m8 :wink:

File: database/db_connection.js (hunk 5 above)
Comment (matthewdking) [reply database/db_connection.js]: this is used in start_server.js

File: src/containers/career.jsx (hunk 9 above)
Comment (matthewdking) [reply src/containers/career.jsx]: [differenceBy](https://lodash.com/docs/4.17.4#differenceBy)

File: src/reducers/data.reducer.js (hunk 10 above)
Comment (matthewdking) [reply src/reducers/data.reducer.js]: its as good as gone

File: src/reducers/index.js (hunk 11 above)
Comment (matthewdking) [reply src/reducers/index.js]: :x:

File: src/store.config.js (hunk 13 above)
Comment (matthewdking) [reply src/store.config.js]: how did I miss this :disappointed:

**Pull request merged**